src/Command/MergeAssetsAndVariationsFilesCommand.php

```diff
@@ -0,0 +1,252 @@
+{
+    protected static $defaultName = 'app:merge-files';
+
+    private const CSV_FIELD_DELIMITER = ';';
+    private const CSV_FIELD_ENCLOSURE = '"';
+    private const CSV_END_OF_LINE_CHARACTER = "\n";
+
+    /** @var SymfonyStyle */
+    private $io;
+
+    /** @var CsvReader */
+    private $assetsReader;
+
+    /** @var CsvReader */
+    private $variationsReader;
+
+    /** @var string[] */
+    private $channels = [];
+
+    /** @var string[] */
+    private $locales = [];
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Merge Assets & Variations CSV files into one')
+            ->addArgument('assetsFilePath', InputArgument::REQUIRED, 'Path to the Assets CSV file')
+            ->addArgument('variationsFilePath', InputArgument::REQUIRED, 'Path to the Variations CSV file path')
+            ->addArgument('targetFilePath', InputArgument::REQUIRED, 'The filePath to the new CSV file to create.')
+        ;
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output)
+    {
+        $this->io = new SymfonyStyle($input, $output);
+
+        $assetsFilePath = $input->getArgument('assetsFilePath');
+        $variationsFilePath = $input->getArgument('variationsFilePath');
+        $targetFilePath = $input->getArgument('targetFilePath');
+
+        $this->io->title('Merge PAM Assets CSV file with PAM Variation CSV file');
+        $this->io->text(
+            sprintf('This command will merge a given PAM Asset CSV file with a given Variations CSV file into one single file: "%s"', $targetFilePath)
+        );
+
+        $hasValidFilePaths = $this->hasValidFilePaths($assetsFilePath, $variationsFilePath);
+        if (!$hasValidFilePaths) {
+            return 1;
+        }
+
+        try {
+            $this->assetsReader = new CsvReader(
+                $assetsFilePath, [
+                    'fieldDelimiter' => self::CSV_FIELD_DELIMITER,
+                    'fieldEnclosure' => self::CSV_FIELD_ENCLOSURE,
+                    'endOfLineCharacter' => self::CSV_END_OF_LINE_CHARACTER,
+                ]
+            );
+
+            $this->variationsReader = new CsvReader(
+                $variationsFilePath, [
+                    'fieldDelimiter' => self::CSV_FIELD_DELIMITER,
+                    'fieldEnclosure' => self::CSV_FIELD_ENCLOSURE,
+                    'endOfLineCharacter' => self::CSV_END_OF_LINE_CHARACTER,
+                ]
+            );
+        } catch (IOException|UnsupportedTypeException|ReaderNotOpenedException $e) {
+            $this->io->error($e->getMessage());
+
+            exit;
+        }
+
+        $this->io->text('');
+
+        $this->io->text('Now merging files and create new Assets...');
+        $this->retrieveChannelsAndLocales();
+        $this->mergeFiles($output, $targetFilePath);
+        $this->io->success(sprintf('%s assets created in "%s"', $this->assetsReader->count(), $targetFilePath));
```

Comment:
$this->io->success('You can now import it directly into your PIM by running the "app:import" command');